CMakeLists.txt

```diff
@@ -9,6 +9,15 @@ cmake_minimum_required(VERSION 3.5)
 
 project(ArrayFire VERSION 3.8.0 LANGUAGES C CXX)
 
+if(NOT CMAKE_VERSION VERSION_LESS 3.9)
```

Comment:
Can you move this to InternalUtils? I don't I want software requirements to be more prominent here and move this type of logic in to the other file.